src/main/java/carpet/commands/DrawCommand.java

```diff
@@ -144,21 +235,269 @@ private static int drawCircle(ServerCommandSource source, BlockPos pos, double r
                 }
             }
         }
-        if (CarpetSettings.fillUpdates)
-        {
+        if (CarpetSettings.fillUpdates) {
 
-            for (BlockPos blockpos1 : list)
-            {
+            for (BlockPos blockpos1 : list) {
                 Block blokc = world.getBlockState(blockpos1).getBlock();
                 world.updateNeighbors(blockpos1, blokc); // or always version????
             }
         }
-        Messenger.m(source, "gi Filled "+affected+" blocks");
+        Messenger.m(source, "gi Filled " + affected + " blocks");
 
         return 1;
     }
-    private static double lengthSq(double x, double y, double z)
-        {
+
+    private static double lengthSq(double x, double y, double z) {
         return (x * x) + (y * y) + (z * z);
     }
+
+    private static int manhattan(int x1, int y1, int z1, int x2, int y2, int z2) {
+        return Math.abs(x1 - x2) + Math.abs(y1 - y2) + Math.abs(z1 - z2);
+    }
+
+    private static int drawDiamond(ServerCommandSource source, BlockPos pos, int radius, BlockStateArgument block,
+    Predicate<CachedBlockPosition> replacement) {
+        return (int) drawDiamond(source, pos, radius, block, replacement, false);
+    }
+
+    private static int drawDiamond(ServerCommandSource source, BlockPos pos, int radius,BlockStateArgument block,
+            Predicate<CachedBlockPosition> replacement, boolean solid) {
+        int affected = 0;
+
+        BlockPos.Mutable mbpos = new BlockPos.Mutable(pos);
+
+        List<BlockPos> list = Lists.newArrayList();
+
+        ServerWorld world = source.getWorld();
+
+        CarpetSettings.impendingFillSkipUpdates = !CarpetSettings.fillUpdates;
+
+        for (int x = -radius; x <= radius; ++x) {
+
+            for (int y = -radius; y <= radius; ++y) {
+
+                for (int z = -radius; z <= radius; ++z) {
+
+                    if (manhattan(0, 0, 0, x, y, z) != radius) {
```

Comment:
See, this is super slow. For 100x100x100 you have to analyze 1M blocks, and only place couple hundreds. It only makes sense to have these custom commands if they are more efficient than simple `/script fill <center> <center-radius> <center+radius> "abs(x)+abs(y)+abs(z)==<radius>" <block> <replace air?>`
Make sure you are only walking the blocks you need to place, like in the sphere drawer above.

Reply:
so shall I remove?

Reply:
actually im working on improving it a ton

Reply:
done: see 7783f60